tornado/test/ioloop_test.py

```diff
@@ -584,6 +585,39 @@ def handle_exception(typ, value, traceback):
         self.assertEqual(self.exception.args[0], "callback")
         self.assertEqual(self.future.exception().args[0], "worker")
 
+    def test_run_in_executor_gen(self):
+        def func(arg1, arg2):
+            return arg1 + arg2
+
+        @gen.coroutine
+        def main():
+            arg1, arg2 = random.random(), random.random()
+            res = yield IOLoop.current().run_in_executor(None, func, arg1, arg2)
+            self.assertEqual(arg1 + arg2, res)
+
+        IOLoop.current().run_sync(main)
+
+    @skipBefore35
+    def test_run_in_executor_native(self):
+        def func(arg1, arg2):
+            return arg1 + arg2
+
+        namespace = exec_test(globals(), locals(), """
+        async def main():
+            arg1, arg2 = random.random(), random.random()
+            res = await IOLoop.current().run_in_executor(None, func, arg1, arg2)
+            self.assertEqual(arg1 + arg2, res)
+        """)
+        IOLoop.current().run_sync(namespace['main'])
+
+    def test_set_default_executor(self):
+        class MyExecutor(futures.ThreadPoolExecutor):
+            pass
+
+        executor = MyExecutor(max_workers=1)
+        IOLoop.current().set_default_executor(executor)
+        self.assertIsInstance(IOLoop.current()._executor, MyExecutor)
```

Comment:
This doesn't test that `set_default_executor` changes the behavior of `run_in_executor`. Try something like the above test, but instead of `other_event.wait()` do `time.sleep(0.01); self.assertFalse(other_event.is_set())`